### PR TITLE
Create Set-AutomateAPIGeneric.ps1

### PR DESCRIPTION
**Pull request by jasonrush**
Created Set-AutomateAPIGeneric.ps1, verified initial functionality

### CODE REVIEW

**Comment by jasonrush** [comment]
Verified on local computer via:
```
$ Set-AutomateAPIGeneric -Page 1 -Endpoint "computers/1111/CommandPrompt?" -Body @{ RunAsAdmin = $false; UsePowerShell = $true; CommandText = "gci 'C:\Users\'"; Directory = "C:\Users\" } -PageSize 5

12345678
```
and
```
$ Get-AutomateAPIGeneric -Endpoint "Computers/1111/CommandExecute/12345678" -AllResults

Id                  : 12345678
ComputerId          : 1111
Command             : @{Id=2; Name=Execute Command}
Status              : Success
Parameters          : {powershell.exe!!! -Command "gci 'C:\Users\'"}
Output              :

                          Directory: C:\Users


                      Mode                LastWriteTime         Length Name

                      ----                -------------         ------ ----

                      d-----        8/29/2019   3:51 AM                Administrator

                      d-r---       11/16/2018   9:04 PM                Public
Fastalk             : False
DateLastInventoried : 2019-10-14T13:24:50
```

**Comment by jasonrush** [comment]
Updated to use System.Web.HttpUtility, and tested with the same test as the original comment. I have not verified paging, however I'm not sure exactly how paging would play into a POST reqeust...

File: Public/Set-AutomateAPIGeneric.ps1 [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        if ($IDs) {
+            $QueryString += "&ids=$IDs"
+        }
+
+        #Expands in the returned object
+        if ($Expand) {
+          $QueryString += "&expand=$Expand"
+        }
+        
+    }
```
[the remark this answers is not in the document]
Comment (jasonrush) [reply Public/Set-AutomateAPIGeneric.ps1]: Updated to use System.Web.HttpUtility to escape parameters.

File: Public/Set-AutomateAPIGeneric.ps1 [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+            $URLNew = "$($url)?page=$($Page)&$QueryString"
+            try {
+                $return = Invoke-RestMethod -Method Post -Uri $URLNew -Headers $script:CWAToken -ContentType "application/json" -Body $BodyString
+            }
+            catch {
+                Write-Error "Failed to perform Invoke-RestMethod to Automate API with error $_.Exception.Message"
+            }
+
+            $ReturnedResults += ($return)
+        }
```
Comment (dkattan) [inline Public/Set-AutomateAPIGeneric.ps1]: Can we dry this up?

It seems to me like we could do a do/while loop where we the condition is like while($AllResults -and $ReturnedResults -and $null -eq $Page) in the condition

**Comment by jasonrush** [comment]
Superseded by issue #36.

**Pull request closed without merging**